Gems/Atom/Asset/ImageProcessingAtom/Code/Source/Processing/ImageConvert.cpp

```diff
@@ -518,6 +531,58 @@ namespace ImageProcessingAtom
         return true;
     }
 
+    // Set (alpha-weighted) average color computed from given mip
+    bool ImageConvertProcess::SetAverageColor(AZ::u32 mip)
+    {
+        // We only work with pixel format rgba32f
+        const EPixelFormat srcPixelFormat = m_image->Get()->GetPixelFormat();
+        if (srcPixelFormat != ePixelFormat_R32G32B32A32F)
+        {
+            AZ_Assert(false, "I only work with pixel format rgba32f");
```

Comment:
I agree with the vagueness of the error message when viewed in isolation. Note that this used to explicitly include the function name, but was changed based on a previous review comment as location and function information automatically get included in Asserts https://github.com/o3de/o3de/pull/7687#discussion_r809427486 -- although it seems that the GUI doesn't always show this extra info. I'll revert it back to that old style with explicit function mention (also because \_\_FUNCTION\_\_ is often more readable than the full AZ_FUNCTION_SIGNATURE anyway).

The restrictions on the pixel format can indeed be relaxed and everything should still work -- at least, it seems, for (uncompressed) file formats that have bitsPerBlock in multiples of 8. The case of a nonlinear encoding would however require some additional code to get the correct linear average, but I'm not sure if additional code to handle that non-linear case is really justified.
Some context: ImageConvertProcess::SetAverageColor() is one in a series of internal steps within the image conversion process that are all handled by private functions (the second step of which is a conversion to rgba32f). In that sense, there's quite a bit of control over the expected input which seems in line with the use of an Assert to detect if that assumption would get broken. I essentially followed the style of the ImageConvertProcess::FillMipmaps() step that happens right before and which shares the same assumptions and asserts. Of course, if it's more fitting, I can transform all those asserts to AZ_Errors.
FWIW, I've made an off-branch commit for a suggestion of SetAverageColor() that is more flexible regarding its input here: https://github.com/RoaldFre/o3de/commit/a86cac2c5b83087eeb64068608baa8ba7558746e (but again, this is not really expected to occur -- and if it were, then other steps in the pipeline will have their assumptions violated as well). I can include it in this branch should it be deemed useful.